File: src/sequence/isCorrectBrackets2/is_correct_brackets2.py
def is_correct_brackets2(input):
    ROUND_BRACKETS = '()'
    SQUARE_BRACKETS = '[]'
    CURLY_BRACKETS = '{}'
    brackets_pairs = [ROUND_BRACKETS, SQUARE_BRACKETS, CURLY_BRACKETS]
    BRACKETS = '([{}])'
    only_brackets = ''.join([ch for ch in input if ch in BRACKETS])
    while ROUND_BRACKETS in only_brackets \
        or SQUARE_BRACKETS in only_brackets \
            or CURLY_BRACKETS in only_brackets:
        for bracket_pair in brackets_pairs:
            if bracket_pair in only_brackets:
                only_brackets = only_brackets.replace(bracket_pair, "")
    result = not only_brackets
    print(result)

File: src/sequence/isCorrectBrackets2/IsCorrectBrackets2.java
package sequence.isCorrectBrackets2;

import java.util.ArrayDeque;
import java.util.Deque;
import java.util.Map;

public class IsCorrectBrackets2 {
    public static final Map<Character, Character> brackets = Map.of(
            '(', ')',
            '[', ']',
            '{', '}');

    public static void isCorrectBrackets2(String input) {
        Deque<Character> stack = new ArrayDeque<>();
        boolean result = true;
        for (int i = 0; i < input.length(); i++) {
            Character current = input.charAt(i);
            if (brackets.containsKey(current)) {
                stack.push(current);
            }
            if (brackets.containsValue(current) && (stack.isEmpty() ||
                    brackets.get(stack.pop()) != current)) {
                result = false;
                break;
            }
        }
        if (result) {
            result = stack.isEmpty();
        }
        System.out.println(result);
    }

}